src/helpers/zapper.js

```diff
@@ -1,49 +1,28 @@
-const zapperApiUrl = 'https://api.zapper.fi/v1';
+import { fetchEventSource } from '@microsoft/fetch-event-source';
+
+const zapperApiUrl = 'https://api.zapper.fi/v2';
 const treasury = '0x3bcf3db69897125aa61496fc8a8b55a5e3f245d5';
-const zapperApiKey = '96e0cc51-a62e-42ca-acee-910ea7d2a241';
+const zapperApiKey = '07809e9c-6d2d-4f6b-ba09-f3f7fb3e1fed';
 
 /* eslint-disable import/prefer-default-export */
 export async function fetchTreasuryBalance() {
-/* eslint-enable import/prefer-default-export */
+  /* eslint-enable import/prefer-default-export */
 
   try {
     let treasuryBalance = 0;
-    const fetchSupported = await fetch(`${zapperApiUrl}/protocols/balances/supported?addresses%5B%5D=${treasury}&api_key=${zapperApiKey}`);
-
-    if (fetchSupported.ok) {
-      const supportedBalances = await fetchSupported.json();
-      const balancesPromises = [];
-
-      supportedBalances.forEach((supportedBalance) => {
-        supportedBalance.apps.forEach((protocol) => {
-          balancesPromises.push(fetch(`${zapperApiUrl}/protocols/${protocol.appId}/balances?addresses%5B%5D=${treasury}&network=${supportedBalance.network}&api_key=${zapperApiKey}`));
-        });
-      });
-
-      return Promise.all(balancesPromises).then(async (balancesResponses) => {
-        for (let i = 0; i < balancesResponses.length; i += 1) {
-          const balanceResponse = balancesResponses[i];
-
-          if (balanceResponse.ok) {
-            /* eslint-disable no-await-in-loop */
-            const specificBalances = await balanceResponse.json();
-            /* eslint-enable no-await-in-loop */
-            const balance = specificBalances[treasury].meta.find((meta) => meta.label === 'Total');
-
-            if (balance) {
-              treasuryBalance += balance.value;
-            }
-          } else {
-            throw new Error(balanceResponse.status);
+    await fetchEventSource(
```

Comment:
Same quesiton, not familiar with how eventSourcing works so assuming there isn't a way to do this natively in JS?